dataflow/gpu-workers/landsat_view.py

```diff
@@ -0,0 +1,227 @@
+
+    # Apply gamma correction.
+    pixels **= 1.0 / gamma
+
+    # Return the pixel values as int8 in the range from 0 to 255,
+    # which is what PIL.Image expects.
+    return scene, tf.cast(pixels * 255.0, dtype=tf.uint8)
+
+
+def save_to_gcs(scene, image, output_path_prefix, format="JPEG"):
+    """Saves a PIL.Image as a JPEG file in the desired path."""
+    filename = os.path.join(output_path_prefix, scene + "." + format.lower())
+    with tf.io.gfile.GFile(filename, "w") as f:
+        image.save(f, format)
+
+
+def run(scenes, output_path_prefix, vis_params, beam_args=None):
+    """Load multiple Landsat scenes and render them as JPEG files."""
+    bands = vis_params["bands"]
+    min_value = vis_params["min"]
+    max_value = vis_params["max"]
+    gamma = vis_params["gamma"]
+
+    options = PipelineOptions(beam_args, save_main_session=True)
+    with beam.Pipeline(options=options) as pipeline:
+        (
+            pipeline
+            | "Create scene IDs" >> beam.Create(scenes)
+            | "Get band paths" >> beam.FlatMap(get_valid_band_paths, bands)
+            | "Load bands" >> beam.MapTuple(load_band)
+            | "Group bands per scene" >> beam.GroupByKey()
+            | "Combine bands to dict"
+            >> beam.MapTuple(lambda scene, bands: (scene, dict(bands)))
+            | "Create RGB pixels"
+            >> beam.MapTuple(lambda scene, bands: (scene, [bands[b] for b in bands]))
+            | "Preprocess pixels"
+            >> beam.MapTuple(preprocess_pixels, min_value, max_value, gamma)
+            | "Convert to image"
+            >> beam.MapTuple(
+                lambda scene, rgb_pixels: (
+                    scene,
+                    Image.fromarray(rgb_pixels.numpy(), mode="RGB"),
+                )
+            )
+            | "Save to Cloud Storage" >> beam.MapTuple(save_to_gcs, output_path_prefix)
+        )
+
+
+if __name__ == "__main__":
+    import argparse
+
+    logging.basicConfig(level=logging.INFO)
+
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--output-path-prefix",
+        required=True,
+        help="Path prefix for output image files. "
+        "This can be a Google Cloud Storage path.",
+    )
+    parser.add_argument(
+        "--scene",
+        dest="scenes",
+        action="append",
+        help="One or more Landsat scene IDs to process, for example "
+        "LC08_L1TP_109078_20200411_20200422_01_T1. "
+        "They must be in the format: "
+        "https://www.usgs.gov/faqs/what-naming-convention-landsat-collections-level-1-scenes",
+    )
+    parser.add_argument(
+        "--bands",
+        nargs=3,
+        default=DEFAULT_RGB_BANDS,
+        help="List of three band names to be mapped to RGB",
+    )
+    parser.add_argument(
+        "--min",
+        type=float,
+        default=DEFAULT_MIN_BAND_VALUE,
+        help="Minimum value of the band value range.",
+    )
+    parser.add_argument(
```

Comment:
Instead of having command line arguments, it is preferred to have TODOs that folks uncomment and customize with their values i.e.
```
def call_sample(): 
   # TODO: Customize these values 
    a = 1
    b = 2
   sample(a, b)

def sample(a, b):
   # ...
```

Reply:
We usually always recommend having command line arguments in Beam pipelines since the runner and other options are usually passed through command line arguments as well. But if you prefer we could have TODOs in the `__main__` and call `run` from there. But that might be inconsistent with how other options are usually passed.

Reply:
+1 to David's response - in Beam examples it's common to have command line arguments, and required pipeline arguments are hard to miss.